tst/SkeletalCartesianClosedOpposite.tst

```diff
@@ -0,0 +1,181 @@
+gap> CocartesianCategoriesTest( finsets, i, a, a, alpha, beta : only_primitive_operations := true );;
+
+gap> CocartesianCategoriesTest( finsets, a, t, i, beta, alpha );;
+gap> CocartesianCategoriesTest( finsets, a, t, i, beta, alpha : only_primitive_operations := true );;
+
+##############################################
+# BraidedCartesianCategoriesTest
+##############################################
+
+gap> a := FinSet( 2 );;
+gap> b := FinSet( 3 );;
+
+gap> BraidedCartesianCategoriesTest( finsets, a, b );;
+gap> BraidedCartesianCategoriesTest( finsets, a, b : only_primitive_operations := true );;
+
+gap> a := FinSet( 3 );;
+gap> b := FinSet( 2 );;
+
+gap> BraidedCartesianCategoriesTest( finsets, a, b );;
+gap> BraidedCartesianCategoriesTest( finsets, a, b : only_primitive_operations := true );;
+
+gap> i := InitialObject( finsets );;
+gap> a := FinSet( 4 );;
+
+gap> BraidedCartesianCategoriesTest( finsets, i, a );;
+gap> BraidedCartesianCategoriesTest( finsets, i, a );;
+
+gap> BraidedCartesianCategoriesTest( finsets, a, i );;
+gap> BraidedCartesianCategoriesTest( finsets, a, i : only_primitive_operations := true );;
+
+gap> t := TerminalObject( finsets );;
+gap> a := FinSet( 3 );;
+
+gap> BraidedCartesianCategoriesTest( finsets, t, a );;
+gap> BraidedCartesianCategoriesTest( finsets, t, a : only_primitive_operations := true );;
+
+gap> BraidedCartesianCategoriesTest( finsets, a, t );;
+gap> BraidedCartesianCategoriesTest( finsets, a, t : only_primitive_operations := true );;
+
+##############################################
+# BraidedCocartesianCategoriesTest
+##############################################
+
+gap> a := FinSet( 2 );;
+gap> b := FinSet( 3 );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, a, b );;
+gap> BraidedCocartesianCategoriesTest( finsets, a, b : only_primitive_operations := true );;
+
+gap> a := FinSet( 3 );;
+gap> b := FinSet( 2 );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, a, b );;
+gap> BraidedCocartesianCategoriesTest( finsets, a, b : only_primitive_operations := true );;
+
+gap> i := InitialObject( finsets );;
+gap> a := FinSet( 4 );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, i, a );;
+gap> BraidedCocartesianCategoriesTest( finsets, i, a : only_primitive_operations := true );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, a, i );;
+gap> BraidedCocartesianCategoriesTest( finsets, a, i : only_primitive_operations := true );;
+
+gap> t := TerminalObject( finsets );;
+gap> a := FinSet( 3 );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, t, a );;
+gap> BraidedCocartesianCategoriesTest( finsets, t, a : only_primitive_operations := true );;
+
+gap> BraidedCocartesianCategoriesTest( finsets, a, t );;
+gap> BraidedCocartesianCategoriesTest( finsets, a, t : only_primitive_operations := true );;
+
+##############################################
+# CartesianClosedCategoriesTest
+##############################################
+
+#@if ValueOption( "no_precompiled_code" ) = false;
```

Comment:
I assume"`= false` does not work. You can use `= fail` or `<> true`.

Reply:
I assume you mean "work" in the sense that the CI now completes? I should elaborate: `ValueOption( "no_precompiled_code" )` should be `fail` (the usual case) or `true` (when testing the non-precompiled code). Thus, the code after `ValueOption( "no_precompiled_code" ) = false` is never executed, which is certainly not what we want.

That's a general thing: one should always make sure that errors actually make the CI fail (which they don't if the code is not even executed). I know a case where the CI was always green and fast, very nice. You can guess from the context why :D

Reply:
I totally didn't think of this. :) Thank you.